apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/Indictment.tsx

```diff
@@ -72,20 +76,78 @@ const Indictment: React.FC = () => {
 
   useDeb(workingCase, ['indictmentIntroduction', 'demands'])
 
+  const setDriversLicenseSuspensionRequest = useCallback(
+    (indictmentCounts?: TIndictmentCount[]) => {
+      let requestDriversLicenseSuspension = false
+
+      const reduce = () =>
```

Comment:
I feel like this could be done in fewer lines of code. I'm thinking if you'd flatten these arrays and then use `any` to check if any offence is `IllegalDrugsDriving`, `PrescriptionDrugsDriving` or `DrunkDriving` with the blood alcohol level thing, that would give you the same results in a lot fewer lines, right?

Reply:
If we flatten he array, we loose the indictment count context which is needed for the `DrunkDriving` case in order to get the blood alcohol content.

The logic goes like this:
- if `DrivingWithoutLicense` is selected in any indictment count, then `false`,
- else if `IllegalDrugDriving` or `PrescriptionDrugsDriving` are selected in any indictment count, then `true`,
- else if `DrunkDriving` is selected in any indictment count with blood alcohol content > 1.2, then `true`.

Reply:
Think I found a way to simplify, by cleaning up the substance map. Will attempt a rewrite.